frontend/dockerfile/linter/ruleset.go

```diff
@@ -41,4 +41,12 @@ var (
 			return fmt.Sprintf("Command '%s' should match the case of the command majority (%s)", violatingCommand, correctCasing)
 		},
 	}
+	RuleMaintainerDeprecated = LinterRule[func() string]{
+		Name:        "MaintainerDeprecated",
+		Description: "The maintainer instruction is deprecated and will generate an error in future versions of the Dockerfile syntax",
```

Comment:
nit: Because we can't actually remove instructions without v2, maybe "The maintainer instruction is deprecated, use a label instead to define image author."